src/NumericalAlgorithms/SphericalHarmonics/IO/ReadSurfaceYlm.cpp

```diff
@@ -27,6 +27,61 @@
 
 namespace ylm {
 namespace {
+const size_t num_non_coef_headers = 5;
+
+// check that the headers of the legend are what we expect them to be
+template <typename Frame>
+void check_legend(const std::vector<std::string>& legend) {
```

Comment:
Instantiate this function as well for all frames?

Reply:
I think we don't have to do this because the instantiations of `read_surface_ylm<Frame>` instantiate `check_legend<Frame>` for all frames since it calls `check_legend<Frame>`. Let me know if you'd still like for me to add instantiations for `check_legend`, despite this!

Reply:
Oh I didn't know nested functions would also be explicitly instantiated. No need to make this change then